Enable/View Controllers/LoginViewController.m

```diff
@@ -0,0 +1,110 @@
+//
+//  LoginViewController.m
+//  Enable
+//
+//  Created by Amanda Wang on 7/6/22.
+//
+
+#import "LoginViewController.h"
+#import "ProfileViewController.h"
+#import "SceneDelegate.h"
+#import "HomeViewController.h"
+#import "Parse/Parse.h"
+#import "UserProfile.h"
+@interface LoginViewController ()
+@property (weak, nonatomic) IBOutlet UITextField *emailTextField;
+@property (weak, nonatomic) IBOutlet UITextField *passTextField;
+@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
+@end
+
+@implementation LoginViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+}
+
+/*
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
+*/
+
+// allow iCloud Keychain ? for future development.
+- (void)navigateToProfile {
+    SceneDelegate *sceneDelegate = (SceneDelegate *)self.view.window.windowScene.delegate;
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+    UINavigationController *navController = [storyboard instantiateViewControllerWithIdentifier:@"mainNav"];
+    sceneDelegate.window.rootViewController = navController;
+    [navController.topViewController performSegueWithIdentifier:@"signedIn" sender:nil];
+}
+- (IBAction)didTapSignUp:(id)sender {
+    NSString *email = self.emailTextField.text;
+    PFUser *user = [PFUser user];
+    user.username = email;
+    user.password = self.passTextField.text;
+    UserProfile * userProfile = [[UserProfile alloc] initWithClassName:@"UserProfile"];
+    userProfile.username = @"Anonymous User";
+    userProfile.email = email;
+    
+    if(![self isEmail:email]){
+        NSLog(@"NOT AN EMAIL !!!!!!!!!!!!!");
+        return;
+    }
+    [user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
+        if (!error) {
+            userProfile.userID = [PFUser currentUser];
+            [userProfile saveInBackgroundWithBlock:^(BOOL saveSucceeded, NSError * _Nullable saveError) {
+                if(!saveError){
+                    [self navigateToProfile];
+                }
+                else {
+                    NSLog(@"issues!");
+                    NSLog(@"%@", saveError.localizedDescription);
+                }
+            }];
+        } else {
+            NSLog(@"👿👿👿👿 doesn't work");
+            NSLog(@"%@", error.localizedDescription);
+        }
+    }];
```

Comment:
I'm nit picking here, but I usually try to have zero business logic in an IBAction (or any UI callback type of function).

So for example here, in `didTapSignUp`, I would grab email and password from the textfields (like you're doing already), and then call another function (let's say `signUp`), that take a email password as input.

It's nice to separate, UI logic and business logic, especially when you start testing. Testing UI functions can be a little difficult so I usually just don't test them BUT all my business logic live in functions easier to test.
For example, you could easily test the `signUp` function passing a wrong email, and test that it is correctly throwing an exception